Anytype/Sources/PresentationLayer/ObjectTypeSearch/ObjectTypeSearchViewModel.swift

```diff
@@ -5,36 +5,59 @@ import Services
 @MainActor
 final class ObjectTypeSearchViewModel: ObservableObject {
     @Published var state = State.searchResults([])
+    @Published var searchText = ""
     
-    private let showPins: Bool
+    let showPins: Bool
     private let showLists: Bool
     
-    private let interactor: ObjectTypeSearchInteractor
+    private let spaceId: String
+    private let workspaceService: WorkspaceServiceProtocol
+    private let typesService: TypesServiceProtocol
+    private let objectTypeProvider: ObjectTypeProviderProtocol
     private let toastPresenter: ToastPresenterProtocol
     
     private let onSelect: (_ type: ObjectType) -> Void
+    private var searchTask: Task<(), any Error>?
     
     nonisolated init(
         showPins: Bool,
         showLists: Bool,
-        interactor: ObjectTypeSearchInteractor,
+        spaceId: String,
+        workspaceService: WorkspaceServiceProtocol,
+        typesService: TypesServiceProtocol,
+        objectTypeProvider: ObjectTypeProviderProtocol,
         toastPresenter: ToastPresenterProtocol,
         onSelect: @escaping (_ type: ObjectType) -> Void
     ) {
         self.showPins = showPins
         self.showLists = showLists
-        self.interactor = interactor
+        self.spaceId = spaceId
+        self.workspaceService = workspaceService
+        self.typesService = typesService
+        self.objectTypeProvider = objectTypeProvider
         self.toastPresenter = toastPresenter
         self.onSelect = onSelect
     }
     
     func search(text: String) {
-        Task {
-            let pinnedTypes = showPins ? try await interactor.searchPinnedTypes(text: text) : []
-            let listTypes = showLists ? try await interactor.searchListTypes(text: text, includePins: !showPins) : []
-            let objectTypes = try await interactor.searchObjectTypes(text: text, includePins: !showPins)
-            let libraryTypes = text.isNotEmpty ? try await interactor.searchLibraryTypes(text: text) : []
-            let defaultType = try interactor.defaultObjectType()
+        searchTask?.cancel()
+        
+        searchTask = Task { @MainActor in
```

Comment:
Your viewmodels is Main Actor. Task can be execute in background. @MainActor is overhead